### PR TITLE
[stable/spinnaker] update spinnaker dependencies

### LINKED ISSUE #20207: [stable/spinnaker] Bug with kubectl 1.16+
Can't install spinnaker's helm release on kubectl v1.16+ 

> helm install --name spinnaker  stable/spinnaker --namespace spinnaker
> Error: validation failed: [unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta2", unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta2"]

kubectl v1.16.4
helm v2.16.1

Reproduce :
Try to install spinnaker stable chart on K8S 1.16+

### PR DESCRIPTION
**Pull request by rrijkse**
#### What this PR does / why we need it:

Updates the requirements file with more recent version to allow deployments on new K8s clusters and updates the hal image to a more recent version.

#### Which issue this PR fixes

  - same as PR #18545 
  - fixes #20207, #18460

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/spinnaker/requirements.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 dependencies:
 - name: redis
-  version: 3.8.0
+  version: 10.5.3
```
Comment (dwardu89) [inline stable/spinnaker/requirements.yaml:3]: Since this is a massive chart change, did you address the possibility of any core changes in the redis chart? There might have been some variables moved round.

File: stable/spinnaker/requirements.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
   repository: https://kubernetes-charts.storage.googleapis.com/
   condition: redis.enabled
 - name: minio
-  version: 1.6.3
+  version: 5.0.9
```
Comment (dwardu89) [inline stable/spinnaker/requirements.yaml:7]: Since this is a massive chart change, did you address the possibility of any core changes in the mini chart? There might have been some variables moved round.

File: stable/spinnaker/requirements.yaml (hunk 1 above)
Comment (rrijkse) [reply stable/spinnaker/requirements.yaml:3]: With the default in values.yaml (which is all I'm using) the redis service/pod deploys without issue.

File: stable/spinnaker/requirements.yaml (hunk 2 above)
Comment (rrijkse) [reply stable/spinnaker/requirements.yaml:7]: Your right I had Minio disabled and it does not deploy (I had it turned off on mine). I committed new defaults that worked in a test environment.

Comment (rrijkse) [reply stable/spinnaker/requirements.yaml:7]: Wanted to add: This is the reason I bumped the minor version number of the chart instead of the patch version, this may require a major version bump depending on what your thoughts are. People will have to update their Minio settings and possibly their Redis settings.

**Comment by rrijkse** [comment]
The error from the end to end test is: `pod didn't trigger scale-up (it wouldn't fit if a new node is added):` not sure there is much I can do about that.

File: stable/spinnaker/values.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ halyard:
   spinnakerVersion: 1.16.1
```
Comment (karlkfi) [inline stable/spinnaker/values.yaml]: Would it make sense to also bump Spinnaker to either the latest patch 1.16.7 or the latest minor 1.18.5?
Or should that go in a different PR?

File: stable/spinnaker/values.yaml [hunk 4]
```diff
@@ -257,11 +257,14 @@ redis:
...
   enabled: true
   image:
-    tag: RELEASE.2018-08-25T01-56-38Z
-  serviceType: ClusterIP
+    tag: RELEASE.2020-01-03T19-12-21Z
+  service:
+    type: ClusterIP
   accessKey: spinnakeradmin
   secretKey: spinnakeradmin
-  bucket: "spinnaker"
```
Comment (karlkfi) [inline stable/spinnaker/values.yaml:264]: You still need the bucket, otherwise you'll have to change `templates/configmap/halyard-configyaml` to read from the default bucket name. Also, I think the latest value syntax is `defaultBuckets: "spinnaker"`, not a map. Per https://github.com/bitnami/charts/blob/master/bitnami/minio/values.yaml#L160

File: stable/spinnaker/values.yaml [hunk 5]
```diff
@@ -257,11 +257,14 @@ redis:
 minio:
   enabled: true
   image:
-    tag: RELEASE.2018-08-25T01-56-38Z
-  serviceType: ClusterIP
+    tag: RELEASE.2020-01-03T19-12-21Z
```
Comment (karlkfi) [inline stable/spinnaker/values.yaml:260]: i think the tag and service type configs have defaults in the upstream bitnami chart. Do we need these overridden?

**Comment by paulczar** [comment]
I think this is needed ... spinnaker has been broken with kube 1.16+ for a while now ...   But there's a bunch of extra stuff to think about ... so I suggest we set the chart version to be a `2.0.0-rc1` and we can iterate a few times on it as a release candidate until we're sure it'll upgrade ok.
